test/malloc_compliance_tests_glibc.cpp

```diff
@@ -0,0 +1,80 @@
+// Copyright (C) 2023 Intel Corporation
+// Under the Apache License v2.0 with LLVM Exceptions. See LICENSE.TXT.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+// run basic compliance tests for glibc's functions (sanity check)
+
+#include "base.hpp"
+
+using umf_test::test;
+
+#include <assert.h>
+#include <stdlib.h>
+
+#include "../src/memory_pool_internal.h"
+#include "malloc_compliance_tests.hpp"
+#include <umf/memory_pool_ops.h>
+
+// casting to (uintptr_t) because of:
+// warning C4312: 'type cast': conversion from 'unsigned int' to 'void *' of greater size
+#define POOL_PRIV ((void *)(uintptr_t)0xDEADBEAF)
+
+static void *glibc_malloc(void *pool, size_t size) {
+    (void)pool; // unused
+    assert(pool == POOL_PRIV);
+    return malloc(size);
+}
+
+static void *glibc_calloc(void *pool, size_t num, size_t size) {
+    (void)pool; // unused
+    assert(pool == POOL_PRIV);
+    return calloc(num, size);
+}
+
+static void *glibc_realloc(void *pool, void *ptr, size_t size) {
+    (void)pool; // unused
+    assert(pool == POOL_PRIV);
+    return realloc(ptr, size);
+}
+
+static enum umf_result_t glibc_free(void *pool, void *ptr) {
+    (void)pool; // unused
+    assert(pool == POOL_PRIV);
+    free(ptr);
+    return UMF_RESULT_SUCCESS;
+}
+
+static const umf_memory_pool_ops_t UMF_GLIBC_POOL_OPS = {
```

Comment:
Please extract this pool to separate files similar to "pool_null" that we have right now. Also please note that glibc - based poll is already created here: https://github.com/oneapi-src/unified-memory-framework/blob/main/test/common/pool.hpp#L112C8. @pbalcer @igchor we should decide to go with C++ approach already used in disjoint pool tests or with this one.  I like this one because of simplicity.. but if you disagree please add more comments to disjoint pool tests because I don't understand this code

Reply:
The `C++` approach is not that complicated. The idea is just to have a class that implements all the functions that are defined in 'ops' structure and that's it. You can then use a helper function to instantiate that pool/provider, like this `umf::poolMakeUnique<umf_test::malloc_pool>(...)`.

The benefit of this is that you avoid a lot of boilerplate and you can easily share the implementation. For example, see this test which creates a custom pool type that has some extra logic in `initialize` method but all other methods have a default behavior (thanks to inheritance): https://github.com/oneapi-src/unified-memory-framework/blob/main/test/memoryPoolAPI.cpp#L160

Reply:
Done